setup.js

```diff
@@ -0,0 +1,184 @@
+const readline = require('readline');
+const fs = require('fs');
+const { execSync } = require('child_process');
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+// Helper function to handle system-specific commands
+function executeSystemCommand(command) {
+  try {
+    execSync(command, { stdio: 'inherit' });
+  } catch (error) {
+    if (error.code === 'ENOENT' || error.code === 'EEXIST') {
+      console.error(error.message);
+      // Try to continue execution even if some non-critical issues occur
+    } else {
+      console.error(error);
+      process.exit(1);
+    }
+  }
+}
+
+rl.question('Name? (This will go on the LICENSE)\n=> ', (name) => {
```

Comment:
Could you remove the nesting here?

Something like this would be optimal:
```js
const question = (str) => new Promise(resolve => rl.question(str, resolve));
const email = question('Email?\n=> ');
// etc..
```

Reply:
Done!